dcos/subcommand.py

```diff
@@ -230,20 +303,31 @@ def _install_env(pkg, options):
     """
 
     pkg_dir = _package_dir(pkg.name())
+    resources = pkg.resource_json()
+    binary_cli = _get_cli_binary(resources)
 
-    install_operation = pkg.command_json()
-
-    env_dir = os.path.join(pkg_dir,
-                           constants.DCOS_SUBCOMMAND_VIRTUALENV_SUBDIR)
-
-    if 'pip' in install_operation:
-        _install_with_pip(
+    if binary_cli:
+        _install_with_binary(
             pkg.name(),
-            env_dir,
-            install_operation['pip'])
+            pkg_dir,
+            binary_cli)
     else:
-        raise DCOSException("Installation methods '{}' not supported".format(
-            install_operation.keys()))
+        install_operation = pkg.command_json()
```

Comment:
`command.json` is optional. So this will fail when `resource.cli` is set but `command.json` is missing. We currently use `has_command_definition` to determine if there is a CLI to install. We need to rework that. One option is to change that method's name to `has_cli_definition` and fix the implementation to look at both `resource.cli` and `command.json`. Even that is not 100% correct. I think the only reason we need that check is so that we can print `Installing CLI subcommand....`. Think about it and let me know how you want to solve this.